### DISCUSSION TITLE
Package listing

### DISCUSSION THREAD

**Discussion by flip111** [flora-pm/flora-server#518] [Feature Requests]
Examples

* https://flora.pm/categories/algorithms
* https://flora.pm/search?q=any

1.
The package name is most important so it just be completely left aligned to be the first thing that you read. This is currently not the case because it is prefixed with "@hackage/" (or similar).

My preference to fix this would be to move "@hackage/" elsewhere. Second solution would be to change the color and/or font-size.

2.
To optimize finding package by looking through the list, the list should not have other items in between the package title. Currently there is license and version "BSD-3-Clausev0.4.0.0". License and version should be moved to the right or go onto the previous line line. There are other pages which list packages such as https://community.chocolatey.org/packages which also have a lot of "noise" between reading the titles. There i don't mind it so much ... why is that?

* On chocolately i mostly need the top result or the one below it. With haskell packges it's more common to wanted to read many results.
* Chocolately arguable has more useful information underneath the title (download, description, by, tags, copy install command) than on flora license + version number

Though i think there is demand and use case for having more information in the listing than would be good to have on a single line.

Proposed solution:
As the first easy fix make the results single line. Remove the license and version number.
As a later implementation add a "expand" or "more info" button that expands the single line results to a box similar to chocolately.

3.
Information that should be in the listing. As already mentioned in my opinion license and version number should not be visible in the single line mode. In expanded view i would keep version number, but not show license either.

I would put direct links to open repository and open documentation. Similar crates.io example: https://crates.io/search?q=json

Also an install command like chocolately is nice nice https://community.chocolatey.org/packages In case of different install commands (cabal/stack). I would put a toggle on the top of the list to change install command `[Cabal / Stack]`

4.
Sorting:

* Alphabetically
* Last version update
* Downloads (when available from statistics)
* Dependents count (as a proxy for popularity)

5.
Filtering
I would make a button or several buttons to filter the search results. (can find an example of how this can look like when necessary). When filtering through the from fields it would be cool to update the search string with the [modifiers](https://flora.pm/documentation/search-features) were applicable. I think gmail for example already has this, you click the "with attachment" button and your search string changes to "has:attachment".

Criteria
Could come up with a lot of criteria to filter by. Different criteria are going to be useful to different people. For example License might be more interesting to commercial users rather than hobbyist.

Personally i would like these criteria:
* In stackage version (also see https://github.com/flora-pm/flora-server/discussions/513 )
* Published after
* Category
* Author
* Dependency

6.
Pagination
Would be nice to have a little bit richer pagination. Perhaps something like this

`PREVIOUS 1, 2, 3, [nr] 10, 20, 30 NEXT`
`182 pages total`

where `[nr]` is the current page number which is an input field to switch page

**Comment by TixieSalander** [comment]
As I said [on another topic ](https://github.com/flora-pm/flora-server/discussions/519#discussioncomment-10839755)you made: very relevant and welcome feedbacks, who will be take into account in the upcoming redesign! 🫶
